doc/rst/developer/bestPractices/buildingdocs.rst

```diff
@@ -1,15 +1,227 @@
+ - RST files in the `docs/rst/` or `chapel/test/release/examples/primers/`
+   subdirectory trees can refer to each other and
+   when they are converted to html, those are converted to links.
+ - The file being linked to has a ``.. _file-tag-name:`` at the top of it.
+ - The file linking to another file uses ``:ref:`file-tag-name``` anywhere
+   in the text.
+ - See https://github.com/chapel-lang/chapel/blob/main/test/release/examples/primers/interopWithC.chpl
+   for an example of ``:ref:`readme-libraries``` that links to
+   the html for
+   https://raw.githubusercontent.com/chapel-lang/chapel/main/doc/rst/technotes/libraries.rst. The rst source file has
+   ``.. _readme-libraries:`` at the top.
+
+
+Linking between files in https://chapel-lang.org/docs/ and
+html files in https://chapel-lang.org .
+
+- Linking from an rst file to an html file directly in https://chapel-lang.org
+  is done by naming the needed URL in the .rst file and using that name.
+  Here are some examples:
+
+``.. _Chapel Developers: https://chapel.discourse.group/c/developers``
+
+``.. _Chapel Users: https://chapel.discourse.group/c/users``
+
+``.. _chapel-lang/chapel: https://github.com/chapel-lang/chapel``
+
+Here is a reference to the ```Chapel Developers`_`` URL later in the .rst file.
+
+- Linking from one of the html files at https://chapel-lang.org such as
+  https://chapel-lang.org/contributing.html can only be done by a core contributor.
+  Post an inquiry to the `Chapel Developers`_ Discourse forum.
+
+
+============================================
+Organization of Chapel Documentation Sources
+============================================
+
+The file chapel/doc/rst/index.rst is the root file that sets up the
+structure of the documentation and results in the current sidebar
+displayed on https://chapel-lang.org/docs/.
+
+
+There is a strong correlation in side bar of
+https://chapel-lang.org/docs/ and the doc/rst/ subdirectories,
+except for lots of the stuff under Writing Chapel Programs.
+
+Below, the various kinds of documentation are organized by how they are
+converted into .html files.  The first column indicates the document name
+in the sidebar of https://chapel-lang.org/docs/, the second column shows
+the source files in the repository, and the last column indicates the sub
+heading the file is under.
+
+.. list-table:: **Using Sphinx to convert .rst files to .html files**
+   :widths: 100 100 100
+   :header-rows: 1
+
+   * - Doc name in Sidebar
+     - Source file(s)
+     - Sidebar subheading
+   * - Quickstart Instructions
+     - doc/rst/usingchapel/QUICKSTART.rst
+     - COMPILING AND RUNNING CHAPEL
+   * - Using Chapel
+     - doc/rst/usingchapel/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Platform-Specific Notes
+     - doc/rst/platforms/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Technical Notes
+     - doc/rst/technotes/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Tools
+     - doc/rst/tools/*
+     - COMPILING AND RUNNING CHAPEL
```

Comment:
Would there be a way to organize this such that the third column was earlier (to reflect the actual hierarchy) without being repetitive, such as:

```
COMPILING AND RUNNING CHAPEL
  QUICKSTART | doc/rst/usingchapel/QUICKSTART.rst
  Using Chapel | doc/rst/usingchapel/*
  Platform-specific notes | doc/rst/platforms/*
  ...
WRITING CHAPEL PROGRAMS
  ...
```

Or if not, maybe just drop the third column?  It took me awhile to even see what it was referring to because I was imagining it to be lower in the hierarchy rather than higher

Reply:
That is how they were organized before but it wasn't working.  The key point here is how the documents are being converted into html.  Where they show up in the sidebar is a secondary point.  So I am going to keep this organization.

Reply:
I don't disagree with your primary vs. secondary point distinction necessarily, but it seems odd to me to have the "doc name in sidebar" and "sidebar subheading" columns (a) not adjacent to each other and (b) not in an order reflecting the hierarchy.  So I'd either permute the columns to 3, 1, 2 or 2, 3, 1 to make sure they're adjacent to each other and to put the major category first.

Reply:
Fixed in PR #18316.